kolibri/logger/utils/user_data.py

```diff
@@ -319,3 +319,49 @@ def add_channel_activity_for_user(**options): # noqa: max-complexity=16
                         masterylog=mastery_log,
                         sessionlog=session_log,
                     )
+
+
+def create_lessons_for_classroom(**options):
+
+    classroom = options['classroom']
+    channels = options['channels']
+    lessons = options['lessons']
+    facility = options['facility']
+    now = options['now']
+
+    coaches = facility.get_coaches()
+    if coaches:
+        coach = random.choice(coaches)
+    else:
+        members = facility.get_members()
+        if not members:
+            coach = FacilityUser.objects.create(username='admin', facility=facility)
+            coach.set_password('password')
+            coach.save()
+        else:
+            coach = random.choice(members)
+            facility.add_coach(coach)
+
+    for l in range(lessons):
+
+        channel = random.choice(channels)
+        channel_content = ContentNode.objects.filter(channel_id=channel.id)
+        # don't add more than 10 resources per Lesson:
+        n_content_items = min(random.randint(0, channel_content.count() - 1), 10)
+        lesson_content = []
+        for i in range(0, n_content_items):
+            # Use this to randomly select a content node to generate the interaction for
+            index = random.randint(0, channel_content.count() - 1)
+            random_node = channel_content[index]
+            content = {"contentnode_id": random_node.id, "channel_id": channel.id, "content_id": random_node.content_id}
+            lesson_content.append(content)
+
+        Lesson.objects.create(
+            title='Lesson {}-{a}'.format(l, a=random.choice('ABCDEF')),
```

Comment:
minor style note - recommend calling this variable `lesson` instead of `l` to improve readability

Reply:
Done